src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,183 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            linkFirst(value);
+        } else {
+            linkLast(value);
+        }
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException(getErrorMessage(index));
+        }
```

Comment:
You already have method to check index, for it to work correctly you can add if clause:
```suggestion
        if (index != size) {
            checkIndex(index);
        }
```